metabase/metabase
# Add card parameter source

Epic https://github.com/metabase/metabase/issues/26898
Demo https://www.loom.com/share/b7de0812f27648dc9326a5ce21e95602
Depends on https://github.com/metabase/metabase/pull/27343

How to verify:
- Add native query `select distinct CATEGORY from PRODUCTS order by CATEGORY`
- Add a dashboard and a question based on products
- Edit dashboard -> Filter -> Text or Category -> Dropdown
- Map this parameter to the question on the dashboard
- Select parameter -> Options to pick from -> From question
- Select the native question and its category column and save the dashboard
- Make sure you can filter the dashboard with dropdown values coming from the question

Notes:
- Only string columns are supported

Not yet implemented:
- Search for questions on the first step
- Empty state for the second step
- Values preview
- Filtering on data types, e.g. for not allowing to select tables. Requires `DataPicker` changes.

<img width="389" alt="Screenshot 2022-12-21 at 22 58 31" src="https://user-images.githubusercontent.com/8542534/209001834-bd3424df-2c05-4cc4-99fd-814e35afe90d.png">
<img width="1133" alt="Screenshot 2022-12-21 at 22 58 46" src="https://user-images.githubusercontent.com/8542534/209001850-1674602f-9cf5-48b2-bffe-b77d6bd5a979.png">
<img width="1143" alt="Screenshot 2022-12-21 at 22 58 53" src="https://user-images.githubusercontent.com/8542534/209001858-91ebb56b-0d34-4727-9cbc-5b5717695305.png">


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/27354)
